SCGF Patrol Craft

LordNest opened the pull request:
Данный PR добавляет патрульный корабль ЦПСС со следующими отличиями от оригинала
* Invider заменён на Z9
* LCD заменена на Z8
* Барабанный дробовик заменён на Узи
* Дробовик помповый заменён на боевой
* Удалён Борон, так как Борон используется только для RUST ГЭК "Факел". (Если ГКК или ЦПСС будут просить сами борон, то бейте их по рукам и по жопе)

close #52 

<details>
<summary>Чейнджлог</summary>

```yml
🆑
rscadd: Добавлена карта SCGF Patrol Craft
maptweak: Удалил лишний АПЦ на карте патруля и добавил новые эмблемы от Вольфора
maptweak: Удалил канистру с бороном
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.

LordNest committed:
Initial Commit

Added SCGF Patrol Ship

LordNest committed:
Second fix commit

LordNest committed:
proceeding killing runtimes

LordNest committed:
Processing...

LordNest committed:
Runtime execution

LordNest committed:
Merge branch 'SierraBay:dev220' into Patrol_ship

SuhEugene commented on mods/maps/sentinel/_map_sentinel.dme:
```diff
@@ -0,0 +1,13 @@
+// END_PREFERENCES
+// BEGIN_INCLUDE
+#include "sentinel.dm"
+// END_INCLUDE
```
```suggestion
#ifndef MODPACK_MAP_SENTINEL
#define MODPACK_MAP_SENTINEL

#include "sentinel.dm"

#endif
```

А также я требую подключения всех файлов карты и составления модпака по [гайду из PR#201](https://github.com/SierraBay/SierraBay12/blob/fda88f7edcb4b4fb640d36c7947d04e44ecd30d8/mods/README.md), который пока что не замёржен.

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,110 @@
+		"nav_patrol_3",
+		"nav_patrol_4",
+		"nav_hangar_reaper"
+		)
```
Красивше
```suggestion
	initial_generic_waypoints = list(
		"nav_patrol_1",
		"nav_patrol_2",
		"nav_patrol_3",
		"nav_patrol_4",
		"nav_hangar_reaper"
	)
```

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,110 @@
+
+	initial_restricted_waypoints = list(
+	"Reaper" = list("nav_hangar_reaper")
+		)
```
ЫАЫЭЫЭАЭЫАЫЫ!!!
```suggestion
	initial_restricted_waypoints = list(
		"Reaper" = list("nav_hangar_reaper")
	)
```

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,110 @@
+	for(var/area/ship/patrol/A)
+		A.name = "\improper [name] - [A.name]"
+		GLOB.using_map.area_purity_test_exempt_areas += A.type
+	..()
```
```suggestion
#define PATROL_SHIP_PREFIX pick("Sentinel","Cavalry","Scarabaeus","Heretic","Apocalypse","Calamatious","Terror","Pandemonium","Anubis","Hound","Stalker","Avatar","Ultimatum","Goliath","Tyrant","Nemesis","Hydra","Stormhawk","Manticore","Basilisk")
/obj/effect/overmap/visitable/ship/patrol/New()
	name = "SFV [PATROL_SHIP_PREFIX], \a [name]"
	for(var/area/ship/patrol/A)
		A.name = "\improper [name] - [A.name]"
		GLOB.using_map.area_purity_test_exempt_areas += A.type
	..()
#undef PATROL_SHIP_PREFIX
```

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,110 @@
+	prefix = "mods/maps/"
+	suffixes = list("sentinel/sentinel-1.dmm", "sentinel/sentinel-2.dmm")
+	spawn_cost = 0.5
+	player_cost = 7 // Нынешнее значение основано на количестве игроков в авейке ~bear1ake
+	spawn_weight = 1 // INF, spawn_weight = 0.5
```
No infinity
```suggestion
	player_cost = 7
	spawn_weight = 1
```

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,110 @@
+/obj/effect/submap_landmark/joinable_submap/patrol
+	name = "Sol Patrol Ship"
+	archetype = /singleton/submap_archetype//away_scg_patrol
+
+//obj/machinery/door/airlock/autoname/command
+	//req_access = list(access_heads)
+
+//obj/machinery/door/airlock/autoname/engineering
+	//req_access = list(access_engine)
+
+//obj/machinery/door/airlock/autoname/marine
+	//req_access = list(access_security)
```
```suggestion
```

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,110 @@
+
+/obj/effect/submap_landmark/joinable_submap/patrol
+	name = "Sol Patrol Ship"
+	archetype = /singleton/submap_archetype//away_scg_patrol
```
Лишний слеш чи шо?
```suggestion
	archetype = /singleton/submap_archetype/away_scg_patrol
```

SuhEugene commented on mods/maps/sentinel/sentinel_areas.dm:
```diff
@@ -0,0 +1,244 @@
+/area/ship/patrol/maintenance/engine
+	icon_state = "engine"
+	ambience = list('sound/ambience/ambigen3.ogg','sound/ambience/ambigen4.ogg','sound/ambience/ambigen5.ogg','sound/ambience/ambigen6.ogg','sound/ambience/ambigen7.ogg','sound/ambience/ambigen8.ogg','sound/ambience/ambigen9.ogg','sound/ambience/ambigen10.ogg','sound/ambience/ambigen11.ogg','sound/ambience/ambieng1.ogg')
+	req_access = list(access_away_cavalry)
```
```suggestion
#define PATROL_ENG_AMBIENCE list('sound/ambience/ambigen3.ogg','sound/ambience/ambigen4.ogg','sound/ambience/ambigen5.ogg','sound/ambience/ambigen6.ogg','sound/ambience/ambigen7.ogg','sound/ambience/ambigen8.ogg','sound/ambience/ambigen9.ogg','sound/ambience/ambigen10.ogg','sound/ambience/ambigen11.ogg','sound/ambience/ambieng1.ogg')
/area/ship/patrol/maintenance/atmos
	name = "\improper Atmospherics Comparment"
	icon_state = "atmos"
	ambience = PATROL_ENG_AMBIENCE
	req_access = list(access_away_cavalry)

/area/ship/patrol/maintenance/engine
	icon_state = "engine"
	ambience = PATROL_ENG_AMBIENCE
	req_access = list(access_away_cavalry)
#undef PATROL_ENG_AMBIENCE
```

SuhEugene commented on mods/maps/sentinel/sentinel_areas.dm:
```diff
@@ -0,0 +1,244 @@
+	icon_state = "yellow"
+	req_access = list(access_away_cavalry)
+
+/*
+/area/ship/patrol/command/lasers
+	name = "\improper Abandoned MRSP platform"
+	icon_state = "yellow"
+	req_access = list(access_away_cavalry)
+*/
+
+
```
```suggestion
```

SuhEugene commented on mods/maps/sentinel/sentinel_crew.dm:
```diff
@@ -0,0 +1,405 @@
+	 Исключением являются те ситуации, когда вы терпите бедствие или на вашем судне аварийная ситуация."
+	required_language = LANGUAGE_HUMAN_EURO
+	whitelisted_species = list(SPECIES_HUMAN)
+	min_skill = list(SKILL_COMBAT  = SKILL_BASIC,
+					 SKILL_WEAPONS = SKILL_BASIC,
+					 SKILL_HAULING = SKILL_BASIC,
+					 SKILL_MEDICAL = SKILL_BASIC,
+					 SKILL_EVA = SKILL_BASIC)
```
```suggestion
	min_skill = list(
		SKILL_COMBAT  = SKILL_BASIC,
		SKILL_WEAPONS = SKILL_BASIC,
		SKILL_HAULING = SKILL_BASIC,
		SKILL_MEDICAL = SKILL_BASIC,
		SKILL_EVA     = SKILL_BASIC
	)
```

SuhEugene commented on mods/maps/sentinel/sentinel_crew.dm:
```diff
@@ -0,0 +1,405 @@
+	\
+	 Вам крайне нежелательно приближаться к кораблям и станциям с опозновательными знаками без разрешения от командования группировкой. \
+	 Исключением являются те ситуации, когда вы терпите бедствие или на вашем судне аварийная ситуация."
+	min_skill = list(SKILL_COMBAT  = SKILL_BASIC,
+					 SKILL_WEAPONS = SKILL_BASIC,
+					 SKILL_HAULING = SKILL_BASIC,
+					 SKILL_MEDICAL = SKILL_BASIC,
+					 SKILL_PILOT = SKILL_ADEPT,
+					 SKILL_EVA = SKILL_BASIC)
```
```suggestion
	min_skill = list(
		SKILL_COMBAT  = SKILL_BASIC,
		SKILL_WEAPONS = SKILL_BASIC,
		SKILL_HAULING = SKILL_BASIC,
		SKILL_MEDICAL = SKILL_BASIC,
		SKILL_PILOT   = SKILL_ADEPT,
		SKILL_EVA     = SKILL_BASIC
	)
```

SuhEugene commented on mods/maps/sentinel/sentinel_crew.dm:
```diff
@@ -0,0 +1,405 @@
+	\
+	 Вам крайне нежелательно приближаться к кораблям и станциям с опозновательными знаками без разрешения от командования группировкой. \
+	 Исключением являются те ситуации, когда вы терпите бедствие или на вашем судне аварийная ситуация."
+	min_skill = list(SKILL_COMBAT  = SKILL_BASIC,
+					 SKILL_WEAPONS = SKILL_BASIC,
+					 SKILL_HAULING = SKILL_ADEPT,
+					 SKILL_MEDICAL = SKILL_EXPERT,
+					 SKILL_ANATOMY = SKILL_BASIC,
+					 SKILL_CHEMISTRY = SKILL_BASIC,
+					 SKILL_EVA = SKILL_BASIC)
```
```suggestion
	min_skill = list(
		SKILL_COMBAT    = SKILL_BASIC,
		SKILL_WEAPONS   = SKILL_BASIC,
		SKILL_HAULING   = SKILL_ADEPT,
		SKILL_MEDICAL   = SKILL_EXPERT,
		SKILL_ANATOMY   = SKILL_BASIC,
		SKILL_CHEMISTRY = SKILL_BASIC,
		SKILL_EVA       = SKILL_BASIC
	)
```

SuhEugene commented on mods/maps/sentinel/sentinel_crew.dm:
```diff
@@ -0,0 +1,405 @@
+	\
+	 Вам крайне нежелательно приближаться к кораблям и станциям с опозновательными знаками без разрешения от командования группировкой. \
+	 Исключением являются те ситуации, когда вы терпите бедствие или на вашем судне аварийная ситуация."
+	min_skill = list(SKILL_COMBAT  = SKILL_BASIC,
+					 SKILL_WEAPONS = SKILL_BASIC,
+					 SKILL_HAULING = SKILL_ADEPT,
+					 SKILL_MEDICAL = SKILL_BASIC,
+					 SKILL_EVA = SKILL_ADEPT,
+					 SKILL_CONSTRUCTION = SKILL_ADEPT,
+					 SKILL_ELECTRICAL = SKILL_ADEPT,
+					 SKILL_ATMOS  = SKILL_BASIC,
+					 SKILL_ENGINES = SKILL_ADEPT,
+					 SKILL_DEVICES = SKILL_BASIC)
```
Бляяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяяя

SuhEugene replied:
```suggestion
	min_skill = list(
		SKILL_COMBAT  = SKILL_BASIC,
		SKILL_WEAPONS = SKILL_BASIC,
		SKILL_HAULING = SKILL_ADEPT,
		SKILL_MEDICAL = SKILL_BASIC,
		SKILL_ATMOS   = SKILL_BASIC,
		SKILL_ENGINES = SKILL_ADEPT,
		SKILL_DEVICES = SKILL_BASIC,
		SKILL_EVA     = SKILL_ADEPT,
		SKILL_ELECTRICAL   = SKILL_ADEPT,
		SKILL_CONSTRUCTION = SKILL_ADEPT,
	)
```

SuhEugene commented on mods/maps/sentinel/sentinel_items.dm:
```diff
@@ -0,0 +1,120 @@
+	name = "SCG armband"
+	desc = "An armband, worn by the crew to display which country they represent. This one is blue and gold."
+	icon_state = "solblue"
+/* MISC
+ * ========
+ */
```
```suggestion
```

SuhEugene commented on mods/maps/sentinel/sentinel_shuttle.dm:
```diff
@@ -0,0 +1,68 @@
+	landmark_tag = "nav_reaper_dock"
+	docking_controller = "nuke_shuttle_dock_airlock"
+
+	////////
+	//TAGS//
+	////////
+
+/*/obj/effect/shuttle_landmark/reaper/hangar
+	name = "Reaper Hangar"
+	landmark_tag = "nav_hangar_reaper"
+	base_area = /area/ship/patrol/command/hangar
+	base_turf = /turf/simulated/floor/plating
+
+/obj/effect/shuttle_landmark/reaper/transit
+	name = "In transit"
+	landmark_tag = "nav_transit_reaper"
+
+/obj/effect/shuttle_landmark/reaper/tdock
+	name = "SEV Torch Dock"
+	landmark_tag = "nav_reaper_dock"
+	docking_controller = "nuke_shuttle_dock_airlock" */
```
Вижу комментарий - советую удалить. keklown.

Или удаляй, или `// SIERRA TODO: сделать чота` писать надо, или ПР всё-же драфт
```suggestion
```

SuhEugene reviewed (changes requested):
Проверил 12 из 16. Дальше лень, потом как-нибудь.

LordNest committed:
Unit_test_canpass

LordNest committed:
Merge branch 'Patrol_ship' of https://github.com/LordNest/SierraBay12 into Patrol_ship

LordNest committed:
and mapfixes

LordNest committed:
Update mods/maps/sentinel/sentinel.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_crew.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_crew.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_shuttle.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_items.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_crew.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_areas.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_areas.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_crew.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/_map_sentinel.dme

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

SuhEugene commented on mods/maps/sentinel/sentinel.dm:
```diff
@@ -0,0 +1,103 @@
+#include "sentinel_areas.dm"
+#include "sentinel_crew.dm"
+#include "sentinel_items.dm"
+#include "sentinel_presets.dm"
+#include "sentinel_shuttle.dm"
+#include "sentinel_structures.dm"
+#include "sentinel_turbolift.dm"
```
В `mods/maps/sentinel/_map_sentinel.dme` давай это дело

LordNest replied:
> В `mods/maps/sentinel/_map_sentinel.dme` давай это дело

так все же карты инициализируют сначала базовый dm а потом уже всякие приколы. 
Не сломается ничего от этого?

SuhEugene commented on mods/maps/sentinel/sentinel_crew.dm:
```diff
@@ -0,0 +1,413 @@
+/var/const/access_away_cavalry_ops = "ACCESS_CAVALRY_OPS"
+/var/const/access_away_cavalry_pilot = "ACCESS_CAVALRY_PILOT"
+/var/const/access_away_cavalry_captain = "ACCESS_CAVALRY_CAPTAIN"
+/var/const/access_away_cavalry_commander = "ACCESS_CAVALRY_COMMANDER"
```
```suggestion
var/global/const/access_away_cavalry = "ACCESS_CAVALRY"
var/global/const/access_away_cavalry_fleet_armory = "ACCESS_CAVALRY_EMERG_ARMORY"
var/global/const/access_away_cavalry_ops = "ACCESS_CAVALRY_OPS"
var/global/const/access_away_cavalry_pilot = "ACCESS_CAVALRY_PILOT"
var/global/const/access_away_cavalry_captain = "ACCESS_CAVALRY_CAPTAIN"
var/global/const/access_away_cavalry_commander = "ACCESS_CAVALRY_COMMANDER"
```

SuhEugene commented on mods/maps/sentinel/sentinel_areas.dm:
```diff
@@ -0,0 +1,238 @@
+/area/ship/patrol/dock
+	name = "\improper Docking Bay"
+	icon_state = "entry_1"
+	req_access = list(access_away_cavalry)
```
Ну как это - файл без пустой строки в конце?
```suggestion
	req_access = list(access_away_cavalry)

```

SuhEugene commented on mods/maps/sentinel/sentinel_turbolift.dm:
```diff
@@ -0,0 +1,25 @@
+	areas_to_use = list(
+		/area/turbolift/sentinel_first,
+		/area/turbolift/sentinel_second
+		)
```
```suggestion
	areas_to_use = list(
		/area/turbolift/sentinel_first,
		/area/turbolift/sentinel_second
	)
```

LordNest committed:
Update mods/maps/sentinel/sentinel_crew.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_turbolift.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_areas.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Merge branch 'SierraBay:dev220' into Patrol_ship

LordNest committed:
Mod

LordNest committed:
Mapfix_mini

LordNest committed:
Unittests_bruteforce

LordNest committed:
Unitology

LordNest committed:
Liberia cleanup

Unit_tests and account removal

LordNest committed:
Unit_tests_pass

LordNest committed:
Merge branch 'SierraBay:dev220' into Patrol_ship

LordNest committed:
Liberia bridge fix

Why this area was even edited? Blast you , infinity coders

LordNest committed:
Merge branch 'Patrol_ship' of https://github.com/LordNest/SierraBay12 into Patrol_ship

LordNest committed:
Area_fix

LordNest committed:
Pipe

Blast it

LordNest committed:
Exterminatus

Due to too many usage of mods and map content, i condemn this unit test to exterminatus

SuhEugene reviewed (changes requested):
Да мне бля в принципе страшно сюда смотреть. Тут много букав читать.
Да не хочу я это ревьювить

SuhEugene commented on maps/away_sites_testing/away_sites_testing.dm:
```diff
@@ -21,6 +21,8 @@
 	#include "../away/miningstation/miningstation.dm"
 	#include "../away/scavver/scavver_gantry.dm"
 	#include "../away/voxship/voxship.dm"
+//	#include "../../mods/maps/liberia/_map_liberia.dme" Uses too many assets from mods and Sierra, which makes separate testing obsolete
+//	#include "../../mods/maps/sentinel/_map_sentinel.dme" Ises too many assets from Torch, which makes separate testing obsolete
```
Оставь здесь такой тег чтобы (мне) потом пофиксить не забыть

```suggestion
// SIERRA TODO: fix that shit
//	#include "../../mods/maps/liberia/_map_liberia.dme" Uses too many assets from mods and Sierra, which makes separate testing obsolete
//	#include "../../mods/maps/sentinel/_map_sentinel.dme" Ises too many assets from Torch, which makes separate testing obsolete
```

SuhEugene commented on mods/maps/liberia/liberia_jobs.dm:
```diff
@@ -15,12 +15,7 @@
 	outfit_type = /singleton/hierarchy/outfit/job/liberia/merchant/leader
 	whitelisted_species = null
 	blacklisted_species = list(SPECIES_VOX, SPECIES_ALIEN, SPECIES_GOLEM, ) // SPECIES_MANTID_GYNE, SPECIES_MANTID_ALATE, SPECIES_MONARCH_WORKER, SPECIES_MONARCH_QUEEN Not yet... not yet...
-	allowed_branches = list(
-		/datum/mil_branch/civilian
-	)
-	allowed_ranks = list(
-		/datum/mil_rank/civ/civ
-	)
```
:achE:

LordNest replied:
Оно наследует от родителя, всё в порядке

SuhEugene commented on mods/maps/sentinel/sentinel_crew.dm:
```diff
@@ -0,0 +1,413 @@
+	\
+	 Вам крайне нежелательно приближаться к кораблям и станциям с опозновательными знаками без разрешения от командования группировкой. \
+	 Исключением являются те ситуации, когда вы терпите бедствие или на вашем судне аварийная ситуация."
+	min_skill = list(SKILL_COMBAT  = SKILL_BASIC,
+					 SKILL_WEAPONS = SKILL_BASIC,
+					 SKILL_HAULING = SKILL_BASIC,
+					 SKILL_MEDICAL = SKILL_BASIC,
+					 SKILL_PILOT = SKILL_ADEPT,
+					 SKILL_EVA = SKILL_BASIC)
```
```suggestion
	min_skill = list(
		SKILL_COMBAT  = SKILL_BASIC,
		SKILL_WEAPONS = SKILL_BASIC,
		SKILL_HAULING = SKILL_BASIC,
		SKILL_MEDICAL = SKILL_BASIC,
		SKILL_PILOT   = SKILL_ADEPT,
		SKILL_EVA     = SKILL_BASIC
	)
```

SuhEugene reviewed (changes requested):
Ладно, это ласт ревью (надеюсь) от меня.
Конечно я хрен знает что тут Либерия забыла, если это сентинель.

Поправь мелкие недочёты и можно поносить в прод.


А ещё хочу чтобы пояснил за мем с удалением бранчей с Либерии в дискордике с пингом

LordNest committed:
Update maps/away_sites_testing/away_sites_testing.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

LordNest committed:
Update mods/maps/sentinel/sentinel_crew.dm

Co-authored-by: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

SuhEugene reviewed (approved):
@ImJustKisik, твой выход
